### DISCUSSION TITLE
Prisma client crash with NextAuth

### DISCUSSION THREAD

**Discussion by Ginn0ji** [nextauthjs/next-auth#5232] [Help]
### Adapter type

@next-auth/prisma-adapter

### Environment

System:
    OS: Windows 10 10.0.19044
    CPU: (6) x64 AMD Ryzen 5 3500X 6-Core Processor
    Memory: 9.08 GB / 15.94 GB

  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD

  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (104.0.1293.63)
    Internet Explorer: 11.0.19041.1566
    I use OperaGX LVL 4 (core: 89.0.4447.64)

  npmPackages:
    next: 12.1.0 => 12.1.0
    next-auth: ^4.10.3 => 4.10.3
    react: 17.0.2 => 17.0.2
    @next-auth/prisma-adapter: ^1.0.4 => 1.0.4
    

### Reproduction URL

https://github.com/Ginn0ji/test

### Describe the issue

Prisma client is crashing with this error: "GetUserByAccountError", when I try signing in using the prisma adapter to push the account, session and user data to my database

### How to reproduce

Install: npm install next-auth @prisma/client @next-auth/prisma-adapter and
npm install prisma --save-dev

Create your pages/api/auth/[...nextauth].js file with any provider you want and a prisma adapter like so:
```JavaScript
import NextAuth from "next-auth"
import GoogleProvider from "next-auth/providers/google"
import { PrismaAdapter } from "@next-auth/prisma-adapter"
import { PrismaClient } from "@prisma/client"

const prisma = new PrismaClient()

export default NextAuth({
  adapter: PrismaAdapter(prisma),
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
    }),
  ],
})
```

Initialize prisma and configure your env file to connect your database

Copy the prisma schema from NextAuth docs and set the correct datasource provider (I'm using sqlite in this example, but I know that postgresql requires a shadowUrl and Planetscale racial integrity, you can easily google that):
```Prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

model Account {
  id                 String  @id @default(cuid())
  userId             String
  type               String
  provider           String
  providerAccountId  String
  refresh_token      String?
  access_token       String? 
  expires_at         Int?
  token_type         String?
  scope              String?
  id_token           String?
  session_state      String?

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}

model Session {
  id           String   @id @default(cuid())
  sessionToken String   @unique
  userId       String
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}

model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  emailVerified DateTime?
  image         String?
  accounts      Account[]
  sessions      Session[]
}

model VerificationToken {
  identifier String
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
}
```

After that make a simple ui like a Sign In button and don't forget to import the signIn function from "next-auth":
```JavaScript
import { signIn } from "next-auth/react"
...
<Button onClick={() => signIn()}>Sign In</Button>
```

Not sure if I forgot something, I hope not.

### Expected behavior

Instead of getting this error, it should've signed me in and then push the account, session and user data to my database.

**Comment by donfour** [comment]
[This fixed it for me.](https://github.com/nextauthjs/next-auth/issues/3815#issuecomment-1035589980)
